chat-server/server.js

```diff
@@ -21,8 +23,68 @@ const welcomeMessage = {
 //We will start with one message in the array.
 const messages = [welcomeMessage];
 
-app.get("/", (request, response) => {
-  response.sendFile(__dirname + "/index.html");
+app.get("/", (req, res) => {
+  res.sendFile(__dirname + "/index.html");
+});
+
+app.post("/messages", (req, res) => {
+  const newMessage = req.body;
+  if (
+    !newMessage.hasOwnProperty("text") ||
+    !newMessage.hasOwnProperty("from") ||
+    newMessage.from === "" ||
+    newMessage.text === "" ||
+    typeof newMessage.text != "string" ||
+    typeof newMessage.from != "string"
+  ) {
+    res.sendStatus(400).send("wrong input");
+  }
+  newMessage["id"] = messages[messages.length - 1].id + 1;
+  newMessage["timeSent"] = new Date();
+  messages.push(newMessage);
+  res.json(messages);
+});
+
+app.get("/messages", (req, res) => {
+  res.json(messages);
+});
+
+app.get("/messages/id/:id", (req, res) => {
+  const messageId = req.params.id;
+  const searchedMessage = messages.find((message) => message.id == messageId);
+  res.json(searchedMessage);
+});
+
+app.delete("/messages/id/:id", (req, res) => {
+  const messageId = req.params.id;
+  const deleteMessage = messages.find((message) => message.id == messageId);
+  messages.splice(messages.indexOf(deleteMessage), 1);
+  res.json(messages);
+});
+
+app.get("/messages/search", (req, res) => {
+  const searchInput = req.query.text.toLowerCase();
+  const filteredMessages = messages.filter((message) =>
+    message.text.toLowerCase().includes(searchInput)
+  );
+  res.json(filteredMessages);
+});
+
+app.get("/messages/latest", (req, res) => {
+  const latestMessages = messages.slice(-10);
```

Comment:
I'm sure you can to this more dynamic instead of using hardcoded length 💯

Reply:
yes it makes more sense. I have updated the route with `/messages/latest/:length` route